src/main/java/infra/rest/location/LocationGatewayImpl.java

```diff
@@ -0,0 +1,41 @@
+package infra.rest.location;
+
+import core.gateway.LocationGateway;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import static infra.util.helper.PropertyValueHelper.getConfigPropValueFromField;
+import static infra.util.validator.IpAddressValidator.isValidIpAddress;
+
+public class LocationGatewayImpl implements LocationGateway {
+
+    private static final String IP_STACK_GET = "http://api.ipstack.com/%s?access_key=%s";
+
+    @Override
+    public String checkLocation(final String ipAddress) throws IOException {
```

Comment:
`checkLocation`  return could be mapped to an Domain object.

```suggestion
    public Location checkLocation(final String ipAddress) throws IOException {
```